models/DataObject/ClassDefinition/Data/Extension/Relation.php

```diff
@@ -63,14 +63,27 @@ protected function getPhpDocClassString($asArray = false)
                 $class[] = '\Pimcore\Model\DataObject\AbstractObject' . $strArray;
             } elseif (is_array($classes)) {
                 foreach ($classes as $item) {
-                    $class[] = sprintf('\Pimcore\Model\DataObject\%s', ucfirst($item['classes']) . $strArray);
+                    $class[] = $this->getMappedClassName('\Pimcore\Model\DataObject\\' . ucfirst($item['classes'])) . $strArray;
                 }
             }
         }
 
         return $class;
     }
 
+    protected function getMappedClassName(string $className): string
+    {
+        try {
+            $className = \Pimcore::getContainer()->get('pimcore.model.factory')->getClassNameFor($className);
+            if ($className[0] !== '\\') {
+                $className = '\\' . $className;
+            }
+        } catch (\Exception) {
```

Comment:
If there is nothing todo in the exception we could also just use finally instead.
https://github.com/pimcore/pimcore/blob/b1275b0a477e5c437f366ed6bf557097626191b5/models/Asset/Document/ImageThumbnail.php#L148

Reply:
And do what in `finally`?

Reply:
Return the `$className` as it would be returned anyways.

Reply:
is changed to finally